ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/WrappingRuleTest.kt

```diff
@@ -1871,6 +1871,628 @@ internal class WrappingRuleTest {
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a multiline semi separated with variables, flow controls and method calls`() {
+            val code =
+                """
+                fun test() {
+                    val a = 0; val b = 0; fun bar() {
+                        // no-op
+                    }; for(i in 0..10) {
+                        println(i); println(i); a++; println(a)
+                    }
+                }
+                """.trimIndent()
+            val formattedCode =
+                """
+                fun test() {
+                    val a = 0;
+                    val b = 0;
+                    fun bar() {
+                        // no-op
+                    };
+                    for(i in 0..10) {
+                        println(i);
+                        println(i);
+                        a++;
+                        println(a)
+                    }
+                }
+                """.trimIndent()
+            wrappingRuleAssertThat(code)
+                .hasLintViolations(
+                    LintViolation(2, 15, "Missing newline after \";\""),
+                    LintViolation(2, 26, "Missing newline after \";\""),
+                    LintViolation(4, 7, "Missing newline after \";\""),
+                    LintViolation(5, 20, "Missing newline after \";\""),
+                    LintViolation(5, 32, "Missing newline after \";\""),
+                    LintViolation(5, 37, "Missing newline after \";\""),
+                ).isFormattedAs(formattedCode)
+        }
+
+        @Nested
+        inner class `Given enum class` {
+            @Test
+            fun `Given a enum without ending semi`() {
+                val code =
+                    """
+                    enum class INDEX1 { ONE, TWO, THREE }
+                    enum class INDEX5 {
+                        ONE,
+                        TWO,
+                        THREE
+                    }
+                    """.trimIndent()
+                wrappingRuleAssertThat(code)
+                    .hasNoLintViolations()
+            }
+
+            @Test
+            fun `Given a enum with ending semi`() {
+                val code =
+                    """
+                    enum class INDEX2 { ONE, TWO, THREE; }
+                    enum class INDEX4 {
+                        ONE, TWO, THREE;
+                    }
+                    enum class INDEX5 {
+                        ONE,
+                        TWO,
+                        THREE;
+                    }
+                    enum class INDEX6 {
+                        ONE,
+                        TWO,
+                        THREE,
+                        ;
+                    }
+                    enum class INDEX7 {
+                        ONE,
+                        TWO,
+                        THREE,
+                        ;
+                        fun foo() = ""
+                    }
```

Comment:
I believe that those example are not relevant. A semicolon followed by a newline is always ignored by the wrapping rule and not specically in enums only. The `INDEX2` (which btw should called `Foo`) the real exception as wrapping the closing brace would result in ugly code after removing the semi-colon.

Reply:
These simply asserting that there is no false +ve here due to semi. `followed by a newline is always ignored by the wrapping rule` this is true only when we know the internal implementation of the `WrappingRule` but in future this assumption can change. Right? Shouldn't we have similar cases in TCs to prevent any break as well?

Reply:
I have updated the name to use `FOO` in enums as well. Also let me know if you don't agree with my above statement(in that case I will remove those TCs as well) and think changing the name or creating TC some-other way can make it more clear